src/components/ChooseLanguage.tsx

```diff
@@ -0,0 +1,103 @@
+import { createForm } from "@modular-forms/solid";
+import { useNavigate } from "@solidjs/router";
+import { createSignal, For, Show } from "solid-js";
+
+import { Button, ExternalLink, InfoBox, NiceP, VStack } from "~/components";
+import { useI18n } from "~/i18n/context";
+import { useMegaStore } from "~/state/megaStore";
+import {
+    Language,
+    eify,
+    timeout,
+    LANGUAGE_OPTIONS,
+    EN_OPTION
+} from "~/utils";
+
+type ChooseLanguageForm = {
+    selectedLanguage: string;
+};
+
+const COMBINED_OPTIONS: Language[] = [...LANGUAGE_OPTIONS];
```

Comment:
I still think we should add in the `EN_OPTION` here so that a user can switch back to english

Reply:
Sure, I forgot to add it.